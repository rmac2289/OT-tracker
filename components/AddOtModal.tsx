import React, { useState, SyntheticEvent } from "react";
import { View, Modal, StyleSheet, TouchableOpacity, Alert } from "react-native";
import * as Calendar from "expo-calendar";
import DateTimePicker from "@react-native-community/datetimepicker";
import { Event } from "../types";
import { calendar_id } from "../config";
import { MonoText } from "./StyledText";
import { theme } from "../constants/Colors";
import { color } from "react-native-reanimated";

interface Props {
  isModalOpen: boolean;
  toggleModal: () => void;
}

const AddOtModal = ({ isModalOpen, toggleModal }: Props) => {
  const [startDate, setStartDate] = useState<Date>(new Date());
  const [endDate, setEndDate] = useState<Date>(new Date());

  const addEvent = async () => {
    try {
      const { status } = await Calendar.requestCalendarPermissionsAsync();
      if (status === "granted") {
        const newEvent = await Calendar.createEventAsync(calendar_id, {
          title: "Ross Forced OT",
          startDate: startDate,
          endDate: endDate,
        });
        console.log(`Event ${newEvent} created.`);
        Alert.alert(`Event added to calendar!`);
      }
    } catch (error) {
      Alert.alert(error.message);
    }
    toggleModal();
  };

  const onChangeStart = (event: Event, selectedDate: Date | undefined) => {
    const currentDate = selectedDate || startDate;
    setStartDate(currentDate);
  };
  const onChangeEnd = (event: Event, selectedDate: Date | undefined) => {
    const currentDate = selectedDate || endDate;
    setEndDate(currentDate);
  };

  return (
    <Modal
      presentationStyle="overFullScreen"
      animationType="slide"
      visible={isModalOpen}
      transparent={true}
    >
      <View style={styles.modalView}>
        <View style={styles.dateInputs}>
          <MonoText style={styles.title}>New Shift</MonoText>
          <View style={styles.individualInput}>
            <MonoText style={styles.dateText}>From</MonoText>

            <DateTimePicker
              testID="dateTimeStart"
              value={startDate}
              // @ts-ignore
              mode="datetime"
              is24Hour={true}
              display="default"
              onChange={onChangeStart}
            />
          </View>
          <View style={styles.individualInput}>
            <MonoText style={styles.dateText}>To</MonoText>
            <DateTimePicker
              testID="dateTimeEnd"
              value={startDate}
              // @ts-ignore
              mode="datetime"
              is24Hour={true}
              display="default"
              onChange={onChangeEnd}
            />
          </View>
        </View>
        <View>
          <TouchableOpacity style={styles.addShiftButton} onPress={addEvent}>
            <MonoText style={styles.buttonText}>Add Shift</MonoText>
          </TouchableOpacity>
          <TouchableOpacity style={styles.closeButton} onPress={toggleModal}>
            <MonoText style={styles.buttonText}>Cancel</MonoText>
          </TouchableOpacity>
        </View>
      </View>
    </Modal>
  );
};

const styles = StyleSheet.create({
  modalView: {
    margin: 20,
    backgroundColor: theme.modalBackground,
    borderRadius: 5,
    padding: 35,
    shadowColor: "#000",
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.25,
    shadowRadius: 4,
    elevation: 5,
    height: "60%",
    justifyContent: "space-between",
    marginTop: "auto",
    marginBottom: "auto",
  },
  title: {
    fontSize: 22,
    fontWeight: "500",
    width: "100%",
    textAlign: "left",
    color: "white",
  },
  dateInputs: {
    display: "flex",
    justifyContent: "center",
    alignItems: "center",
  },
  individualInput: {
    width: "100%",
    marginTop: 20,
  },
  dateText: {
    fontSize: 18,
    fontWeight: "700",
    color: "white",
  },
  closeButton: {
    width: "100%",
    backgroundColor: theme.button,
    height: 60,
    marginTop: 10,
    padding: 5,
    alignItems: "center",
    justifyContent: "center",
    borderRadius: 3,
    shadowColor: "#000",
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.25,
    shadowRadius: 4,
    elevation: 5,
  },
  addShiftButton: {
    width: "100%",
    backgroundColor: theme.itemtext,
    height: 60,
    marginTop: 10,
    padding: 5,
    alignItems: "center",
    justifyContent: "center",
    borderRadius: 3,
    shadowColor: "#000",
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.25,
    shadowRadius: 4,
    elevation: 5,
  },
  buttonText: {
    color: "white",
    fontSize: 20,
    fontWeight: "700",
  },
});

export default AddOtModal;
