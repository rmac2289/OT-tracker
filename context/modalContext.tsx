import React, { createContext, useState } from "react";

interface Props {}

export const ModalContext = createContext<any>(null);

export const ModalContextProvider = ({
  children,
}: {
  children: JSX.Element;
}) => {
  const [fadeBackground, setFadeBackground] = useState(false);

  return (
    <ModalContext.Provider value={[fadeBackground, setFadeBackground]}>
      {children}
    </ModalContext.Provider>
  );
};
