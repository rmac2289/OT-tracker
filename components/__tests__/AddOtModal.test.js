import React from "react";
import renderer from "react-test-renderer";
import { ModalContextProvider } from "../../context/modalContext";
import AddOtModal from "../AddOtModal";

it("renders correctly", () => {
  const tree = renderer
    .create(
      <ModalContextProvider>
        <AddOtModal />
      </ModalContextProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
