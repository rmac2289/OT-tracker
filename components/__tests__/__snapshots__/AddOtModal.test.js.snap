// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Modal
  animationType="slide"
  hardwareAccelerated={false}
  presentationStyle="overFullScreen"
  transparent={true}
  visible={true}
>
  <View
    style={
      Object {
        "backgroundColor": "rgba(45, 40, 62,0.9)",
        "borderRadius": 5,
        "elevation": 5,
        "height": "60%",
        "justifyContent": "space-between",
        "margin": 20,
        "marginBottom": "auto",
        "marginTop": "auto",
        "padding": 35,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.25,
        "shadowRadius": 4,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000",
            },
            Array [
              Object {
                "color": "white",
                "fontSize": 22,
                "fontWeight": "500",
                "textAlign": "left",
                "width": "100%",
              },
              Object {
                "fontFamily": "Menlo",
              },
            ],
          ]
        }
      >
        New Shift
      </Text>
      <View
        style={
          Object {
            "marginTop": 20,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000",
              },
              Array [
                Object {
                  "color": "white",
                  "fontSize": 18,
                  "fontWeight": "700",
                },
                Object {
                  "fontFamily": "Menlo",
                },
              ],
            ]
          }
        >
          From
        </Text>
      </View>
      <View
        style={
          Object {
            "marginTop": 20,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000",
              },
              Array [
                Object {
                  "color": "white",
                  "fontSize": 18,
                  "fontWeight": "700",
                },
                Object {
                  "fontFamily": "Menlo",
                },
              ],
            ]
          }
        >
          To
        </Text>
      </View>
    </View>
    <View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#362247",
            "borderRadius": 3,
            "elevation": 5,
            "height": 60,
            "justifyContent": "center",
            "marginTop": 10,
            "opacity": 1,
            "padding": 5,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.25,
            "shadowRadius": 4,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000",
              },
              Array [
                Object {
                  "color": "white",
                  "fontSize": 20,
                  "fontWeight": "700",
                },
                Object {
                  "fontFamily": "Menlo",
                },
              ],
            ]
          }
        >
          Add Shift
        </Text>
      </View>
      <View
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#a4545f",
            "borderRadius": 3,
            "elevation": 5,
            "height": 60,
            "justifyContent": "center",
            "marginTop": 10,
            "opacity": 1,
            "padding": 5,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.25,
            "shadowRadius": 4,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000",
              },
              Array [
                Object {
                  "color": "white",
                  "fontSize": 20,
                  "fontWeight": "700",
                },
                Object {
                  "fontFamily": "Menlo",
                },
              ],
            ]
          }
        >
          Cancel
        </Text>
      </View>
    </View>
  </View>
</Modal>
`;
